### ISSUE TITLE
Remove deprecated support for passing strings instead of keys in conch

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7054]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Alex reported|
|-|-|
|Trac ID|trac#7054|
|Type|enhancement|
|Created|2014-03-25 18:36:41Z|
|Branch|https://github.com/twisted/twisted/tree/remove-string-keys-7054|

This has been deprecated since like 2008.

<details><summary>Searchable metadata</summary>

```
trac-id__7054 7054
type__enhancement enhancement
reporter__Alex Alex
priority__normal normal
milestone__ 
branch__branches_remove_string_keys_7054 branches/remove-string-keys-7054
branch_author__radix radix
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1395772601000000 1395772601000000
changetime__1396032336000000 1396032336000000
version__None None
owner__Alex Alex

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|Alex commented|
|-|-|

This is addressed by: https://github.com/twisted/twisted/pull/9

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_3"></a>|@radix commented|
|-|-|

(In [42020]) Branching to remove-string-keys-7054.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_4"></a>|@radix set owner to Alex|
|-|-|

Looks reasonable, +1.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_5"></a>|@radix set status to `closed`|
|-|-|

(In [42023]) Merge remove-string-keys-7054

Author: alex
Reviewer: radix
Fixes: [#7054](https://github.com/twisted/twisted/issues/7054)

Remove deprecated code that handled public and private
keys in string format.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

This change produces a rather interesting behavior for code that hasn't been updated to avoid the deprecation warning (or new code that accidentally returns a string).

It silently squashes the value to `None` and proceeds as if no key is available.

It seems to me that's not the best thing to happen.
